### ISSUE TITLE
Large difference between reported PaddleOCR performance and converted PyTorch model performance

### ISSUE THREAD

**Issue by duguyue100** [frotms/PaddleOCR2Pytorch#28]
Hi, thanks for releasing this nice converter, it's been very useful! I'm trying to benchmark the performance of PaddleOCR model and the corresponding converted PyTorch model on ICDAR2015 dataset and found there is a big gap between them that I couldn't explain, can you help me?

The model I'm using is `det_r50_vd_db_v2.0_train` and the corresponding PyTorch model is `det_r50_vd_db_v2.0_infer.pth`.

The reported performance for `DB ResNet50_vd` is precision: 86.41%, recall: 78.72%, hmean 82.38%.

However, the best performance I can get from the converted model so far is precision: 66.51%, recall: 69.14%, hmean: 67.80%.
The metric is calculated using the PaddleOCR's `DetMetric` class and I've matched exactly the same preprocessing and post-processing pipelines. I also excluded those bounding boxes which have `###` as labels as done in the PaddleOCR data preparation.

I've also searched issues in PaddleOCR and see if others experienced performance degradation and adjust some inference parameters such as using `image_shape` as the same as in the training configuration file.

Do you have an idea what's going on in my case? Am I missing something? I would love to have your advice.

Yuhuang.

**Comment by frotms** [comment]
@duguyue100 
The possible reasons：
1. Is there a gap between DetMetric and ICDAR 2015 official metrics？
3. Is there any difference between report and code? Test the performance on paddle model using PaddleOCR.
4. The inference parameters, pre-processing method and post-processing method have been changed several times with PaddleOCR. I am not sure which one got the performance high. Ask PaddleOCR for more details
5. There's a gap between PyTorch and Paddle usages and operators in some hard case maybe, or I use some PyTorch interfaces incorrectly

**Comment by duguyue100** [comment]
I ran the evaluation code using Paddle (spent surprisingly big effort, :)), and `precision`, `recall`, and `hmean` match with the reported number in the table.

I'm still trying to figure out what's happening in my PyTorch code. I see your code uses the same preprocessing and post-processing pipelines, I compared the code between PaddleOCR and PaddleOCR2PyTorch, they should be the same.

There is no way that the weights got damaged when copying over to the PyTorch model right?

Thanks for your reply, and I will keep you posted.

Yuhuang.

**Comment by frotms** [comment]
I got almost the same inference results of some images with a lot of boundingboxes by using PaddleOCR2PyTorch and PaddleOCR inference model. The PaddleOCR2PyTorch results are on the left, the PaddleOCR results are on the right. The differences are almost invisible.
![Snipaste_2022-05-04_13-26-24](https://user-images.githubusercontent.com/18061450/166627261-05cb0398-902f-4ab2-9976-551b94756d07.jpg)
![Snipaste_2022-05-04_13-27-06](https://user-images.githubusercontent.com/18061450/166627273-46d2f77d-73af-41e9-b179-5b97579ea565.jpg)
![Snipaste_2022-05-04_13-27-19](https://user-images.githubusercontent.com/18061450/166627275-cbc07c7f-716e-4d2e-83e8-03ae4f16d040.jpg)
![Snipaste_2022-05-04_13-27-36](https://user-images.githubusercontent.com/18061450/166627279-84bf5dc3-494f-4e19-891d-ca2bbf00368f.jpg)
![Snipaste_2022-05-04_13-27-55](https://user-images.githubusercontent.com/18061450/166627281-9aa9fa9d-87e7-446d-aa4c-c8233e2d75e8.jpg)
![Snipaste_2022-05-04_13-28-09](https://user-images.githubusercontent.com/18061450/166627284-a33f28c4-5038-475d-aa41-843849b864be.jpg)
![Snipaste_2022-05-04_13-28-27](https://user-images.githubusercontent.com/18061450/166627287-411a840f-1e8e-4f99-8174-c1e87b6445d0.jpg)

**Comment by duguyue100** [comment]
Thanks a lot for your reply, it's very helpful to see visual examples.

OK, I've found the problem in my evaluation code. The problem is with how I treated `###` labels in the evaluation.
Instead of labeling the `ignore_tags` as `True` as in the original code, I exclude them directly from the data generation.
Now I get 
```
{'precision': 0.8148854961832062, 'recall': 0.8223399133365431, 'hmean': 0.818595734483585}
```
And everything works fine.

Thanks a lot for your help!

Yuhuang.
